### PR TITLE
Balder/wire in wand range

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst @toregge PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/fef/indexproperties.h [hunk 1]
```diff
@@ -178,6 +178,18 @@ namespace mutate {
 
 // Add temporary flags used for safe rollout of new features here
 namespace temporary {
+/**
+ * A number in the range [0,1] for the effective idf range for WeakAndOperator.
+ * 1.0 will give the complete range as used by default by bm25.
+ * idf = (1.0 - range) + (range * idf)
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/fef/indexproperties.h:184]: The formula is missing a detail: `scaled_idf = (1.0 - range) * max_idf + (range * idf)`

File: searchlib/src/vespa/searchlib/queryeval/wand/wand_parts.h [hunk 2]
```diff
@@ -412,6 +407,38 @@ struct TermFrequencyScorer
...
+    using Bm25Executor = features::Bm25Executor;
+    Bm25TermFrequencyScorer(uint32_t num_docs, float range) noexcept
+        : _num_docs(num_docs),
+          _range(range),
+          _max_idf(Bm25Executor::calculate_inverse_document_frequency(1, _num_docs))
+    { }
+    double apply_range(double idf) const noexcept {
+        return (1.0 - _range)*_max_idf + _range * idf;
+    }
+    // weight * idf, scaled to fixedpoint
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/wand/wand_parts.h:422]: I think this needs an update.

**Review by geirst — DISMISSED** [review]
👍

File: searchlib/src/vespa/searchlib/fef/indexproperties.h (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/fef/indexproperties.h]: Fixed

File: searchlib/src/vespa/searchlib/queryeval/wand/wand_parts.h (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/wand/wand_parts.h]: Fixed

**Pull request merged**